packages/core/src/2d/sprite/Sprite.ts

```diff
@@ -100,52 +92,93 @@ export class Sprite extends RefObject {
     const x = MathUtil.clamp(value.x, 0, 1);
     const y = MathUtil.clamp(value.y, 0, 1);
     this._atlasRegionOffset.set(x, y, MathUtil.clamp(value.z, 0, 1 - x), MathUtil.clamp(value.w, 0, 1 - y));
-    this._setDirtyFlagTrue(DirtyFlag.positions | DirtyFlag.uv);
+    this._dispatchSpriteChange(SpritePropertyDirtyFlag.atlas);
+  }
+
+  /**
+   * The rectangle region of the sprite, specified in normalized.
+   */
+  get region(): Rect {
+    return this._region;
+  }
+
+  set region(value: Rect) {
+    const region = this._region;
+    const x = MathUtil.clamp(value.x, 0, 1);
+    const y = MathUtil.clamp(value.y, 0, 1);
+    region.set(x, y, MathUtil.clamp(value.width, 0, 1 - x), MathUtil.clamp(value.height, 0, 1 - y));
+    this._dispatchSpriteChange(SpritePropertyDirtyFlag.atlas);
   }
 
   /**
    * Location of the sprite's center point in the rectangle region, specified in normalized.
+   * The origin is at the bottom left and the default value is (0.5, 0.5).
    */
   get pivot(): Vector2 {
     return this._pivot;
   }
 
   set pivot(value: Vector2) {
     const pivot = this._pivot;
-    const { x, y } = value;
-    if (pivot === value || pivot.x !== x || pivot.y !== y) {
-      pivot.set(x, y);
-      this._setDirtyFlagTrue(DirtyFlag.positions);
+    if (pivot === value) {
+      this._dispatchSpriteChange(SpritePropertyDirtyFlag.pivot);
+    } else {
+      const { x, y } = value;
+      if (pivot.x !== x || pivot.y !== y) {
+        pivot.set(x, y);
+        this._dispatchSpriteChange(SpritePropertyDirtyFlag.pivot);
+      }
     }
   }
```

Comment:
Write like this can solve `sprite.pivot.x=2` not work problem
```ts
set pivot(value: Vector2) {
    const pivot = this._pivot;
    if (pivot !== value) {
      pivot.copyFrom(value);
    }

 constructor() {
    //@ts-ignore
    this._position._onValueChanged = this._onPivotChanged = this._onPivotChanged.bind(this);
  }

private _onPivotChanged(): void {
    this._dispatchSpriteChange(SpritePropertyDirtyFlag.pivot);
  }

```